### ISSUE TITLE
Connection loss that isn't NoConnection doesn't lead to account getting relogged

### ISSUE THREAD

**Issue by HerrEurobeat** [HerrEurobeat/steam-comment-service-bot#95]
**Describe the bug**
A connection loss of an account to Steam which isn't because of NoConnection (eresult 3) doesn't trigger a relog. Instead the message
```
Couldn't log in bot9 after 2 attempt(s). Error 20: Error: ServiceUnavailable
Failed account is not bot0. Skipping account...
```
is shown. The account will remain offline and cause an error when trying to comment.

**Additional context**
This is caused because after disabling the `autoRelogin` feature the `error` event instead of the `disconnected` event is called on a connection loss. The error event doesn't yet know how to distinguish a connection loss other than eresult 3 from an initial startup login failure. Not sure why it shows 2 failed attempts but this is ultimately irrelevant because a failed relog should be retried many more times unless the error is a critcal error (like invalidPassword, logOnSessionReplaced etc.).

**Comment by HerrEurobeat** [comment]
I rewrote (or more like restructured) the error event so disconnects & login errors should be handled correctly now.

**Comment by HerrEurobeat** [comment]
Fixed in 560e130 (Version 2.10.6).
